### DISCUSSION TITLE
How to get error.data in RTK Query using typescript?

### DISCUSSION THREAD

**Discussion by kuldeepmi7874** [community/community#20662] [Projects and Issues] [answered]
If I output error to the console, I get:
```
{
    status: 401,
    data: "Invalid password"
}
```
If I try to output error.data to the console, I get an error:
```
const [login, { data, error }] = useLoginMutation()

const handleSubmit: React.FormEventHandler<HTMLFormElement> = async (event) => {
    event.preventDefault()

    if (credential.user_email.trim() !== '' && credential.user_password.trim() !== '') {
        await login(credential)
    }
}

useEffect(() => {
    if (data !== undefined) {
        localStorage.token = data.token
    }
}, [data])

useEffect(() => {
    if (error !== undefined) {
        console.log(error.data)
    }
}, [error])
```

**Comment by chiragmehta900** [comment, marked as answer]
At this point, it could be an error from the server (`FetchBaseQueryError`) or just any error thrown by code you wrote (`SerializedError`, e.g. in `query`, `queryFn`, `transformResponse `etc.) - and that could have a completely different shape.

To make sure it's a `FetchBaseQueryError`, just do


```
if ('data' in error) {
  // TypeScript will handle it as `FetchBaseQueryError` from now on.
}
```

**Reply by laeeqFaiz** [reply]
how about handling the SerializedError?

**Comment by unknowVariable** [comment]
The error you're getting is likely because the error object doesn't have a data property, so trying to access error.data will result in an error.

Based on the error object that you provided, it seems like data and status are properties of the error object. So if you want to log the error message to the console, you can try console.log(error.data) which should be Invalid password. However, if you want to log the entire error object, you can try console.log(error) instead.
